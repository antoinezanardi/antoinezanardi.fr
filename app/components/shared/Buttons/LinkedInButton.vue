<template>
  <a
    class="btn btn-icon btn-info btn-lg btn-round"
    data-bs-toggle="tooltip"
    :href="ANTOINE_ZANARDI_LINKEDIN_URL"
    rel="noopener noreferrer"
    target="_blank"
    :title="$t('LinkedInButton.joinMyNetwork')"
  >
    <WrappedFontAwesomeIcon
      icon="fa-brands fa-linkedin-in"
      icon-color="white"
      size="2x"
    />
  </a>
</template>

<script setup lang="ts">
import WrappedFontAwesomeIcon from "~/components/shared/Icons/WrappedFontAwesomeIcon/WrappedFontAwesomeIcon.vue";
import { ANTOINE_ZANARDI_LINKEDIN_URL } from "~/shared/constants/antoine-zanardi.constants";
</script>