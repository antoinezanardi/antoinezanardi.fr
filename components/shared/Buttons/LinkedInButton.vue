<template>
  <a
    class="btn btn-info btn-round btn-lg btn-icon"
    target="_blank"
    rel="noopener noreferrer"
    href="https://www.linkedin.com/in/antoinezanardi/"
    data-bs-toggle="tooltip"
    :title="$t('LinkedInButton.joinMyNetwork')"
  >
    <WrappedFontAwesomeIcon icon="fa-brands fa-linkedin-in" size="2x" icon-color="#00000"/>
  </a>
</template>

<script setup lang="ts">
import WrappedFontAwesomeIcon from "~/components/shared/WrappedFontAwesomeIcon.vue";
</script>