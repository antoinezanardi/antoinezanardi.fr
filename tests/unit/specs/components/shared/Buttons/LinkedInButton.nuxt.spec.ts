import type { mount } from "@vue/test-utils";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";

import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";

import LinkedInButton from "@/components/shared/Buttons/LinkedInButton.vue";

describe("LinkedIn Button Component", () => {
  let wrapper: ReturnType<typeof mount<typeof LinkedInButton>>;

  async function mountLinkedInButtonComponent(options: ComponentMountingOptions<typeof LinkedInButton> = {}):
  Promise<ReturnType<typeof mount<typeof LinkedInButton>>> {
    return mountSuspendedComponent(LinkedInButton, { ...options });
  }

  beforeEach(async() => {
    wrapper = await mountLinkedInButtonComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });
});