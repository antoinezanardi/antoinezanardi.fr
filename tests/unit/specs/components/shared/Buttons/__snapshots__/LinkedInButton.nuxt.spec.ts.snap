// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LinkedIn Button Component > should match snapshot when rendered. 1`] = `
"<a data-v-d82ba599="" class="btn btn-icon btn-info btn-lg btn-round" data-bs-toggle="tooltip" href="https://www.linkedin.com/in/antoinezanardi/" rel="noopener noreferrer" target="_blank" title="Rejoignez mon rÃ©seau">
  <wrapped-font-awesome-icon-stub data-v-d82ba599="" icon="fa-brands fa-linkedin-in" iconcolor="white" size="2x" classes=""></wrapped-font-awesome-icon-stub>
</a>"
`;
